rust-toolchain.toml

```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2022-10-22"
+channel = "nightly-2023-06-01"
```

Comment:
I think it is worth upgrading here even though we can't upgrade the enclave workspace toolchain

Reply:
well, I was wrong. this breaks tests. had to revert.
Still worth noting that the main build actually works with the new 1.70 rustc toolchain